addons/survey/static/tests/tours/survey_form.js

```diff
@@ -0,0 +1,268 @@
+        in_modal: true,
+    },
+    ...changeTab("options"),
+    {
+        content: "Change trigger question",
+        trigger: ".o_field_widget[name=triggering_question_id] input",
+        run: "click",
+        in_modal: true,
+    }, {
+        content: "Set the first question as trigger instead",
+        trigger: 'ul.ui-autocomplete a:contains("Question 1")',
+        run: 'click',
+        in_modal: true,
+    }, {
+        content: "Check that the alert is gone",
+        trigger: `.o_form_sheet_bg div:first-child:not(.alert-warning).o_form_sheet`,
+        in_modal: true,
+        run: function () {}, // it's a check
+    }, {
+        content: "Change the trigger answer as well",
+        trigger: ".modal-content .o_field_widget[name=triggering_answer_id] input",
+        run: "click",
+        in_modal: true,
+    }, {
+        content: "Set the second answer as trigger, then",
+        trigger: 'ul.ui-autocomplete a:contains("Answer B")',
+        run: 'click',
+    },
+    ...saveAndClose(),
+     {
+        content: "Check that Question 3 has its 'normal' trigger icon back",
+        trigger: "tr.o_data_row td[data-tooltip='Question 3'] + td + td div.o_widget_survey_question_trigger button i.fa-code-fork",
+        run: function () {}, // it's a check,
+    }, {
+        content: "Move Question 3 back below Question 2",
+        trigger: "tr.o_data_row:nth-child(2) td[name=sequence]",
+        run: "drag_and_drop_native div[name=question_and_page_ids] table tbody tr:nth-child(3)",
+    }, {
+        content: "Open that question again",
+        trigger: "tr.o_data_row td[data-tooltip='Question 3']",
+        run: "click",
+    },
+    ...changeTab("options"),
+    {
+        content: "Change trigger to confirm we can now use Question 2 again",
+        trigger: ".modal-content .o_field_widget[name=triggering_question_id] input",
+        run: "click",
+        in_modal: true,
+    }, {
+        content: "Question 2 is allowed as trigger again",
+        trigger: '.modal-content ul.ui-autocomplete a:contains("Question 2")',
+        in_modal: true,
+    }, {
+        content: "Change the trigger answer back as well",
+        trigger: ".modal-content .o_field_widget[name=triggering_answer_id] input",
+        in_modal: true,
+    }, {
+        content: "Set the second answer as trigger, then",
+        trigger: '.modal-content ul.ui-autocomplete a:contains("Answer B")',
+        run: "click",
+    },
+    ...saveAndClose(),
+    // delete this test survey
+    {
+        content: "Open Action menu to delete the test survey",
+        trigger: ".o_cp_action_menus button",
+    }, {
+        content: "Delete the test survey",
+        trigger: ".dropdown-menu span:contains('Delete')",
+        run: "click",
+    }, {
+        content: "Confirm deletion",
+        trigger: "button.btn-primary:contains('Ok')",
+        run: "click",
+    }
```

Comment:
It's ran inside a python tour, so that should not be necessary, data will be cleaned after the test automatically.